core/src/com/unciv/ui/worldscreen/unit/UnitActions.kt

```diff
@@ -296,6 +301,66 @@ object UnitActions {
                 }.takeIf { unit.currentMovement > 0 && canPillage(unit, tile) })
     }
 
+    private fun pillageLooting(tile: TileInfo, unit: MapUnit) {
+        // Stats objects for reporting pillage results in a notification
+        val pillageYield = Stats()
+        val globalPillageYield = Stats()
+        val toCityPillageYield = Stats()
+        val closestCity = unit.civInfo.cities.minByOrNull { it.getCenterTile().aerialDistanceTo(tile) }
+        val improvement = tile.ruleset.tileImprovements[tile.improvement]!!
+
+        for (unique in improvement.getMatchingUniques(UniqueType.PillageYieldRandom)) {
+            for (stat in unique.stats) {
+                val looted = Random.nextInt((stat.value + 1).toInt()) + Random.nextInt((stat.value + 1).toInt())
+                pillageYield.add(stat.key, looted.toFloat())
+            }
+        }
+        for (unique in improvement.getMatchingUniques(UniqueType.PillageYieldFixed)) {
+            for (stat in unique.stats) {
+                pillageYield.add(stat.key, stat.value)
+            }
+        }
+
+        for (stat in pillageYield) {
+            when (stat.key) {
+                Stat.Gold, Stat.Faith, Stat.Science, Stat.Culture -> {
+                    unit.civInfo.addStat(stat.key, stat.value.toInt())
+                    globalPillageYield[stat.key] += stat.value
+                }
+                Stat.Food, Stat.Production -> {
+                    if (closestCity != null) {
+                        closestCity.addStat(stat.key, stat.value.toInt())
+                        toCityPillageYield[stat.key] += stat.value
+                    }
+                }
+                else -> {}
+            }
+        }
+
+        if (toCityPillageYield.values.sum() > 0 && closestCity != null) {
```

Comment:
Stats.isEmpty() is the same as values.sum() > 0 **if** you're sure not to have negative values in there.

Reply:
Done.